macros/answerHints.pl

```diff
@@ -111,6 +123,11 @@ sub AnswerHints {
     my $student = $ans->{student_value};
     Value::Error("AnswerHints can only be used with MathObjects answer checkers") unless ref($correct);
     return $ans unless ref($student);
+    my $context = $correct->context;
+    my $hash = $context->{answerHash};
+    $context->{answerHash} = $ans;
+    my $processPreview = $correct->getFlag('answerHintsProcessPreview', 0);
+    $context->{answerHash} = $hash;
```

Comment:
I am not sure what this is all about.  I am certain that line 129 is needed.  I am also certain that line 128 is not needed.  I don't know about the rest.  They certainly aren't needed for what is done in this pull request.  What are they for?

Reply:
> Is the point to temporarily ensure the correct answer hash is used for the getFlag call?

Yes, that is correct.  Because these flags can be set on the MathObject, the `cmp()`, or the context itself, the `getFlag()` function looks through all these places to find if the flag is set.  The `$context->{answerHash}` is where `getFlag()` gets access to the flags in the `cmp()` call (which have been added to the answerHash earlier).

> I am also certain that line 128 is not needed

It really is needed, as you have figured out.  All these lines are required.